### PR TITLE
[Security] Fix tests in 2.8

### PR DESCRIPTION
**Pull request by wouterj**
ping @fabpot

### CODE REVIEW

**Comment by fabpot** [comment]
Thank you @WouterJ.

**Comment by fabpot** [comment]
Tests are still broken for me :(

**Comment by stof** [comment]
@WouterJ please check your changes. The tests you modified here are failing

**Comment by WouterJ** [comment]
@stof @fabpot LegacyAbstractVoterTest should be removed, forgot to do it in this PR. I cannot do it before this evening, so feel free to do it yourself.

**Comment by fabpot** [comment]
That's not enough to fix the tests:

```
PHPUnit 4.8.6 by Sebastian Bergmann and contributors.

...............................................................  63 / 572 ( 11%)
............................................................... 126 / 572 ( 22%)
.....................................FFFFF....F................ 189 / 572 ( 33%)
............................................................... 252 / 572 ( 44%)
............................................................... 315 / 572 ( 55%)
............................................................... 378 / 572 ( 66%)
............................................................... 441 / 572 ( 77%)
............................................................... 504 / 572 ( 88%)
............................................................... 567 / 572 ( 99%)
.....

Time: 6.32 seconds, Memory: 36.25Mb

There were 6 failures:

1) Symfony\Component\Security\Core\Tests\Authorization\Voter\AbstractVoterTest::testVoteLegacy with data set #0 (array('EDIT'), 1, stdClass Object (), 'ACCE
SS_GRANTED if attribute a...access')
ACCESS_GRANTED if attribute and class are supported and attribute grants access
Failed asserting that 0 matches expected 1.

src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php:66

2) Symfony\Component\Security\Core\Tests\Authorization\Voter\AbstractVoterTest::testVoteLegacy with data set #1 (array('CREATE'), -1, stdClass Object (), 'A
CCESS_DENIED if attribute an...access')
ACCESS_DENIED if attribute and class are supported and attribute does not grant access
Failed asserting that 0 matches expected -1.

src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php:66

3) Symfony\Component\Security\Core\Tests\Authorization\Voter\AbstractVoterTest::testVoteLegacy with data set #2 (array('DELETE', 'EDIT'), 1, stdClass Object
 (), 'ACCESS_GRANTED if one attribu...access')
ACCESS_GRANTED if one attribute is supported and grants access
Failed asserting that 0 matches expected 1.

src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php:66

4) Symfony\Component\Security\Core\Tests\Authorization\Voter\AbstractVoterTest::testVoteLegacy with data set #3 (array('DELETE', 'CREATE'), -1, stdClass Obj
ect (), 'ACCESS_DENIED if one attribut...access')
ACCESS_DENIED if one attribute is supported and denies access
Failed asserting that 0 matches expected -1.

src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php:66

5) Symfony\Component\Security\Core\Tests\Authorization\Voter\AbstractVoterTest::testVoteLegacy with data set #4 (array('CREATE', 'EDIT'), 1, stdClass Object
 (), 'ACCESS_GRANTED if one attribu...access')
ACCESS_GRANTED if one attribute grants access
Failed asserting that 0 matches expected 1.

src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php:66

6) Symfony\Component\Security\Core\Tests\Authorization\Voter\AbstractVoterTest::testNoOverriddenMethodsThrowsException
Failed asserting that exception of type "\BadMethodCallException" is thrown.

FAILURES!
Tests: 572, Assertions: 1214, Failures: 6.

Legacy deprecation notices (15)
```

File: src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php [hunk 1]
```diff
@@ -54,13 +54,48 @@ public function testVote(array $attributes, $expectedVote, $object, $message)
...
+            && in_array($attribute, array('EDIT', 'CREATE'));
+    }
+}
+
+class AbstractVoterTest_LegacyVoter extends AbstractVoter
+{
     protected function getSupportedClasses()
     {
-        return array('stdClass');
+        return array('AbstractVoterTest_Object');
```
Comment (WouterJ) [inline src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php]: This change should be reverted, it should accept stdClass.

Comment (WouterJ) [inline src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php]: Afaics, this should fix the tests (but I'm on a mobile now, sp can't be 100% sure).

**Pull request merged**